oshi-core/src/main/java/oshi/util/ExecutingCommand.java

```diff
@@ -116,11 +116,33 @@ private ExecutingCommand() {
      *         string if the command failed
      */
     public static List<String> runNative(String[] cmdToRunWithArgs, String[] envp) {
+        Process p = null;
         try {
-            Process p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
+            p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
             return getProcessOutputAndDestroy(p, cmdToRunWithArgs);
         } catch (SecurityException | IOException e) {
             LOG.trace("Couldn't run command {}: {}", Arrays.toString(cmdToRunWithArgs), e.getMessage());
+        } finally {
+            if (p != null) {
+                if (p.getOutputStream() != null) {
+                    try {
+                        p.getOutputStream().close();
```

Comment:
If `exec()` succeeded, `p` is destroyed and it has no output stream to close.

Reply:
This is not true, In Solaris after succesfull execution of exec() I can see open pipes that are opened. After I added this enhancement problem solved. I am calling OSHI every 5 minutes and i can see the increase y using ls -al /proc/${PID}/fd. the p.destory() does not close these pipes. did you realy tried it on Solaris? I can show it to you in our Solaris.

Reply:
I'd be happy to review the logs and any information you have

1. Are you using the latest OSHI _**snapshot**_ version (including the #1767 fix referenced)?  If not, I acknowledge the bug in the latest _released_ version but think it's been fixed by #1767.
2. Please let me know exactly what JVM you are using as OpenJDK has dropped Solaris support so I do not know whether your port properly handles this case.